sql-cli/sql_cli/connections.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import yaml
+from airflow.api_connexion.exceptions import BadRequest
+from airflow.api_connexion.schemas.connection_schema import connection_schema
+from airflow.models import Connection
+from airflow.utils.session import create_session
+from marshmallow.exceptions import ValidationError
+
+from sql_cli.settings import SQL_CLI_PROJECT_DIRECTORY
+
+
+def _load_yaml_connections(environment: str) -> list[dict[str, Any]]:
+    """Gets the configuration yaml for the given environment and loads the connections from it into a dictionary"""
+    config_file = (
+        SQL_CLI_PROJECT_DIRECTORY / "config" / environment / "configuration.yaml"
+    )
+    if not config_file.exists():
+        raise FileNotFoundError(
+            f"Config file configuration.yaml does not exist for environment {environment}"
+        )
+
+    connections = yaml.safe_load(open(config_file))["connections"]
```

Comment:
Please don't use `open()` like this - instead please use the context manager to open and close the file reliably. In case of a crash the file might not be closed. [This](https://betterprogramming.pub/context-managers-in-python-go-beyond-with-open-as-file-85a27e392114) describes the difference quite well.